skglm/experimental/quantile_huber.py

```diff
@@ -0,0 +1,203 @@
+            residual = y[i] - Xw[i]
+            grad_j += -X[i, j] * self._grad_per_sample(residual)
+        return grad_j / n_samples
+
+    def _grad_per_sample(self, residual):
+        """Calculate gradient for a single sample."""
+        tau = self.quantile
+        delta = self.delta
+        r = residual
+
+        if r >= delta:
+            # Upper linear tail: r >= delta
+            return tau
+        elif r >= 0:
+            # Upper quadratic: 0 <= r < delta
+            return tau * r / delta
+        elif r > -delta:
+            # Lower quadratic: -delta < r < 0
+            return (1 - tau) * r / delta
+        else:
+            # Lower linear tail: r <= -delta
+            return tau - 1
+
+    def get_lipschitz(self, X, y):
+        n_features = X.shape[1]
+
+        lipschitz = np.zeros(n_features, dtype=X.dtype)
+        c = max(self.quantile, 1 - self.quantile) / self.delta
+        for j in range(n_features):
+            lipschitz[j] = c * (X[:, j] ** 2).sum() / len(y)
+
+        return lipschitz
+
+    def get_global_lipschitz(self, X, y):
+        c = max(self.quantile, 1 - self.quantile) / self.delta
+        return c * norm(X, ord=2) ** 2 / len(y)
+
+    def intercept_update_step(self, y, Xw):
+        n_samples = len(y)
+
+        # Compute gradient
+        grad = 0.0
+        for i in range(n_samples):
+            residual = y[i] - Xw[i]
+            grad -= self._grad_per_sample(residual)
+        grad /= n_samples
+
+        # Apply step size 1/c
+        c = max(self.quantile, 1 - self.quantile) / self.delta
+        return grad / c
+
+
+class SmoothQuantileRegressor(BaseEstimator, RegressorMixin):
+    """Quantile regression with progressive smoothing."""
+
+    def __init__(self, quantile=0.5, alpha=0.1, delta_init=1.0, delta_final=1e-3,
+                 n_deltas=10, max_iter=1000, tol=1e-6, verbose=False,
+                 fit_intercept=True):
+        self.quantile = quantile
+        self.alpha = alpha
+        self.delta_init = delta_init
+        self.delta_final = delta_final
+        self.n_deltas = n_deltas
+        self.max_iter = max_iter
+        self.tol = tol
+        self.verbose = verbose
+        self.fit_intercept = fit_intercept
+
+    def fit(self, X, y):
+        """Fit using progressive smoothing: delta_init --> delta_final."""
+        w = np.zeros(X.shape[1])
+        deltas = np.geomspace(self.delta_init, self.delta_final, self.n_deltas)
+
+        if self.verbose:
+            print(
+                f"Progressive smoothing: delta {self.delta_init:.2e} --> "
+                f"{self.delta_final:.2e} in {self.n_deltas} steps")
+
+        datafit = QuantileHuber(quantile=self.quantile, delta=self.delta_init)
+        penalty = L1(alpha=self.alpha)
+
+        # Use AndersonCD solver
+        solver = AndersonCD(max_iter=self.max_iter, tol=self.tol,
+                            warm_start=True, fit_intercept=self.fit_intercept,
+                            verbose=3)
```

Comment:
this should be `verbose=max(0,verbose - 1)`   (so that we pass verbose >= 2 corresponds to inner solver being verbose)